packages/api/src/resolvers/mutations/filters/createFilter.ts

```diff
@@ -1,20 +1,23 @@
 import { log } from '@unchainedshop/logger';
 import { Root, Context } from '@unchainedshop/types/api.js';
-import { Filter } from '@unchainedshop/types/filters.js';
+import { Filter, FilterInputText } from '@unchainedshop/types/filters.js';
 
 export default async function createFilter(
   root: Root,
-  { filter }: { filter: Filter & { title: string } },
+  { filter, texts }: { filter: Filter; texts: Array<FilterInputText> },
   context: Context,
 ) {
   const { modules, localeContext, userId } = context;
   log('mutation createFilter', { userId });
 
-  return modules.filters.create(
+  const newFilter = await modules.filters.create(
     {
       ...filter,
+      title: '',
       locale: localeContext.language,
     },
     context,
   );
+  await modules.filters.texts.updateTexts({ filterId: newFilter._id }, texts);
+  return modules.filters.findFilter({ filterId: newFilter._id });
```

Comment:
You can savely return newFilter here (no need for extra request) because the texts are stored in another collection.